From 33d097772438f346419963e7136f65b498dafe22 Mon Sep 17 00:00:00 2001
From: someone5678 <nemui3353@gmail.com>
Date: Sat, 12 Aug 2023 17:01:41 +0800
Subject: [PATCH 13/13] core: Hide ADB and developer setting enable status

* Some apps check whether ADB or developer setting is enabled
* Return 0 to selected apps when those apps getInt
  ADB_ENABLED, ADB_WIFI_ENABLED, DEVELOPMENT_SETTINGS_ENABLED

mnri: move to getIntForUser to cover more scenarios
Change-Id: Ibbf92c7b1f1181d2508cb22ff2fc2017f9e59706
Signed-off-by: minaripenguin <minaripenguin@users.noreply.github.com>
[dereference23: Simplified the code]
Signed-off-by: Alexander Winkowski <dereference23@outlook.com>
---
 core/java/android/provider/Settings.java | 41 ++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/core/java/android/provider/Settings.java b/core/java/android/provider/Settings.java
index 0dd963457..087e2a05b 100644
--- a/core/java/android/provider/Settings.java
+++ b/core/java/android/provider/Settings.java
@@ -4497,6 +4497,9 @@ public final class Settings {
         /** @hide */
         @UnsupportedAppUsage
         public static int getIntForUser(ContentResolver cr, String name, int def, int userHandle) {
+            if (shouldHideSetting(cr.getPackageName(), name)) {
+                return 0;
+            }
             String v = getStringForUser(cr, name, userHandle);
             return parseIntSettingWithDefault(v, def);
         }
@@ -4528,6 +4531,9 @@ public final class Settings {
         @UnsupportedAppUsage
         public static int getIntForUser(ContentResolver cr, String name, int userHandle)
                 throws SettingNotFoundException {
+            if (shouldHideSetting(cr.getPackageName(), name)) {
+                return 0;
+            }
             String v = getStringForUser(cr, name, userHandle);
             return parseIntSetting(v, name);
         }
@@ -7375,6 +7381,9 @@ public final class Settings {
         /** @hide */
         @UnsupportedAppUsage
         public static int getIntForUser(ContentResolver cr, String name, int def, int userHandle) {
+            if (shouldHideSetting(cr.getPackageName(), name)) {
+                return 0;
+            }
             String v = getStringForUser(cr, name, userHandle);
             return parseIntSettingWithDefault(v, def);
         }
@@ -7405,6 +7414,9 @@ public final class Settings {
         /** @hide */
         public static int getIntForUser(ContentResolver cr, String name, int userHandle)
                 throws SettingNotFoundException {
+            if (shouldHideSetting(cr.getPackageName(), name)) {
+                return 0;
+            }
             String v = getStringForUser(cr, name, userHandle);
             return parseIntSetting(v, name);
         }
@@ -18596,6 +18608,9 @@ public final class Settings {
          * or not a valid integer.
          */
         public static int getInt(ContentResolver cr, String name, int def) {
+            if (shouldHideSetting(cr.getPackageName(), name)) {
+                return 0;
+            }
             String v = getString(cr, name);
             return parseIntSettingWithDefault(v, def);
         }
@@ -18620,6 +18635,9 @@ public final class Settings {
          */
         public static int getInt(ContentResolver cr, String name)
                 throws SettingNotFoundException {
+            if (shouldHideSetting(cr.getPackageName(), name)) {
+                return 0;
+            }
             String v = getString(cr, name);
             return parseIntSetting(v, name);
         }
@@ -21791,4 +21809,27 @@ public final class Settings {
         }
         return packages[0];
     }
+
+    /**
+     * @hide
+     */
+    public static boolean shouldHideSetting(@NonNull String packageName, @NonNull String name) {
+        switch (name) {
+            case Settings.Global.ADB_ENABLED:
+            case Settings.Global.ADB_WIFI_ENABLED:
+            case Settings.Global.DEVELOPMENT_SETTINGS_ENABLED:
+                return !isCallerSystemApp(packageName);
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * @hide
+     */
+    private static boolean isCallerSystemApp(@NonNull String packageName) {
+        return packageName.equals("android") ||
+               packageName.startsWith("com.android.") ||
+               packageName.equals("com.google.android.settings.intelligence");
+    }
 }
-- 
2.51.0

